default/python/universe_generation/monsters.py

```diff
@@ -1,8 +1,110 @@
+import sys
 import random
 import freeorion as fo
 import util
 import statistics
 import universe_tables
+from galaxy import DisjointSets
+
+
+def populate_monster_fleet(fleet_plan, system):
+    """
+    Create a monster fleet in ''system'' according to ''fleet_plan''
+    """
+
+    # create monster fleet
+    monster_fleet = fo.create_monster_fleet(system)
+    # if fleet creation fails, report an error and try to continue with next system
+    if monster_fleet == fo.invalid_object():
+        errstr = "Python generate_monsters: unable to create new monster fleet %s" % fleet_plan.name()
+        util.report_error(errstr)
+        raise RuntimeError(errstr)
+
+    # add monsters to fleet
+    for design in fleet_plan.ship_designs():
+        # create monster, if creation fails, report an error and try to continue with the next design
+        if fo.create_monster(design, monster_fleet) == fo.invalid_object():
+            errstr = "Python generate_monsters: unable to create monster %s" % design
+            util.report_error(errstr)
+            raise RuntimeError(errstr)
+
+    print "Spawn", fleet_plan.name(), "at", fo.get_name(system)
+
+monsters_that_alter_starlanes = {"SM_EXP_OUTPOST"}
+
+
+def place_experimentors(systems):
+    """
+    Place the expermentor outpost system.  Check that removing all of the
+    starlanes entering the system does not break the rest of the galaxy
+    into more than one connected piece.
+    """
+
+    random.shuffle(systems)
+
+    # Find the Experimentor output fleet plan
+    plans = [plan for plan in fo.load_monster_fleet_plan_list()
+             if (plan.name() in monsters_that_alter_starlanes
+                 and plan.spawn_rate() > 0.0
+                 and plan.spawn_limit() > 0
+                 and plan.ship_designs())]
+    if not plans:
+        return
+
+    plan = plans[0]
```

Comment:
Correct me if I'm wrong, but won't this code break if someone decides to script another starlane altering monster and just adds its name to the `monsters_that_alter_starlanes`, expecting this to be sufficient to work? Especially since most of the code you wrote here seems to be suited to allow for multiple kinds of starlane altering monsters...